base/float.jl

```diff
@@ -850,21 +850,13 @@ truncmask(x, mask) = x
 
 ## Array operations on floating point numbers ##
 
-float(A::AbstractArray{<:AbstractFloat}) = A
-
-function float(A::AbstractArray{T}) where T
-    if !isleaftype(T)
-        error("`float` not defined on abstractly-typed arrays; please convert to a more specific type")
-    end
-    convert(AbstractArray{typeof(float(zero(T)))}, A)
-end
-
-float(r::StepRange) = float(r.start):float(r.step):float(last(r))
-float(r::UnitRange) = float(r.start):float(last(r))
-float(r::StepRangeLen) = StepRangeLen(float(r.ref), float(r.step), length(r), r.offset)
-function float(r::LinSpace)
-    LinSpace(float(r.start), float(r.stop), length(r))
-end
+# Optimizations for broadcasting float()
+broadcast(::typeof(float), A::AbstractArray{<:AbstractFloat}) = A
```

Comment:
No problem, I'm not sure what's better here either.  At the moment I went for the faster version, as it allows broadcast to be a drop in replacement for vectorized `float()`.

Do we have any precedent for not returning a copy of the data in `broadcast()`.  I had a quick look through `Base`, and all I found was was that `broadcast(::typeof(xor), B::BitArray, x::Bool)` takes the opposite stance and calls `copy()` when it wouldn't otherwise be required.